habitat/tasks/nav/greedy_follower.py

```diff
@@ -0,0 +1,164 @@
+                else SIM_NAME_TO_ACTION[SimulatorActions.RIGHT.value]
+            )]
+            self._reset_agent_state(current_state)
+
+            # Check if forward reduces geodesic distance
+            curr_dist = self._geo_dist(goal_pos)
+            self._sim.step(SIM_NAME_TO_ACTION[SimulatorActions.FORWARD.value])
+            new_dist = self._geo_dist(goal_pos)
+            new_pos = self._sim.get_agent_state().position
+            movement_size = np.sqrt(np.sum(np.square(new_pos - current_state.position)))
+            self._reset_agent_state(current_state)
+            if new_dist < curr_dist and movement_size / self._step_size > 0.99:
+                return action_list_to_one_hot(SIM_NAME_TO_ACTION[SimulatorActions.FORWARD.value])
+            else:
+                return action_list_to_one_hot(best_turn)
+
+    def _reset_agent_state(self, state: habitat_sim.AgentState) -> None:
+        self._sim.set_agent_state(state.position, state.rotation, reset_sensors=False)
+
+    @staticmethod
+    def _angle_between_quats(q1: np.quaternion, q2: np.quaternion) -> float:
+        q1_inv = np.conjugate(q1)
+        dq = quaternion.as_float_array(q1_inv * q2)
+
+        return 2 * np.arctan2(np.linalg.norm(dq[1:]), np.abs(dq[0]))
+
+    @staticmethod
+    def _quat_from_two_vectors(v0: np.array, v1: np.array) -> np.quaternion:
+        v0 = v0 / np.linalg.norm(v0)
+        v1 = v1 / np.linalg.norm(v1)
+        c = v0.dot(v1)
+        if c < (-1 + 1e-8):
+            c = max(c, -1)
+            m = np.stack([v0, v1], 0)
+            _, _, vh = np.linalg.svd(m, full_matrices=True)
+            axis = vh.T[:, 2]
+            w2 = (1 + c) * 0.5
+            w = np.sqrt(w2)
+            axis = axis * np.sqrt(1 - w2)
+            return np.quaternion(w, *axis)
+
+        axis = np.cross(v0, v1)
+        s = np.sqrt((1 + c) * 2)
+        return np.quaternion(s * 0.5, *(axis / s))
+
+    def _geo_dist(self, goal_pos: np.array) -> float:
+        return self._sim.geodesic_distance(self._sim.get_agent_state().position, goal_pos)
+
+    def _est_max_grad_dir(self, goal_pos: np.array) -> np.array:
+
+        current_pos = self._sim.get_agent_state().position
+
+        if self.mode == "spath":
+            points = self._sim.straight_spath_points(self._sim.get_agent_state().position, goal_pos)
+            # NB: Add a little offset as things get weird if
+            # points[1] - points[0] is anti-parallel with forward
+            if len(points) < 2:
+                return None
+            max_grad_dir = self._quat_from_two_vectors(
+                self._sim.FORWARD,
+                points[1]
+                - points[0]
+                + 1e-3 * np.cross(self._sim.UP, self._sim.FORWARD),
+            )
+        else:
+            current_rotation = self._sim.get_agent_state().rotation
+            current_dist = self._geo_dist(goal_pos)
+
+            best_delta = -1e10
```

Comment:
If we can change naming to give more meaning to `delta` variable, that can help to maintain this code.